compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -763,6 +763,9 @@ impl<'rt, 'tcx, M: Machine<'tcx>> ValidityVisitor<'rt, 'tcx, M> {
                 }
                 interp_ok(true)
             }
+            ty::Pat(base_ty, ..) => self.try_visit_primitive(
+                &value.transmute(self.ecx().layout_of(*base_ty)?, self.ecx())?,
```

Comment:
Primitive types basically by definition do not have fields, so having recursion here is very odd. This should probably be handled elsewhere.

Does the pattern not affect validity?

Reply:
it does, and validity checking is still handling it accidentally because all use cases have the pattern type wrapped in a struct 🤦

Reply:
gonna pull this all out into a separate PR with separate tests